### ISSUE TITLE
Not working with CAGE-seq bam files

### ISSUE THREAD

**Issue by DimitrisGrig** [erhard-lab/iTiSS#1]
Hello,

I am trying to utilize this implementation on CAGE-seq bam files. 

I am running the following command:
```
java -cp iTiSS.jar -Xmx12g -Xms4g executables/TiSS -prefix analysis/K562 -bams K562.bam -chromSizes /mnt/raid0/dgrigor/annotation/GRCh38chrom.sizes -modType DENSE_PEAK -autoparam
```

And get the following error:

```
java.lang.RuntimeException: Could not run gedi.TiSSController@4bbefdef
Could not run gedi.TiSSController@4bbefdef
```

[https://i.imgur.com/oEltnYG.png](url)

Is this implementation suitable for CAGE libraries?

**Comment by Chrizey91** [comment]
Hi @DimitrisGrig, sorry for responding so late. I did not get any notification that a new issue was created...

Yes, iTiSS is suitable for CAGE libraries! I will extend the info text in the coming update to make this clear.

Regarding your problem: Could you run the exact same command again but add a `-D` at the end of it? This way, a debug message is printed when the error occurs. If you could copy this here, I will gladly help you fix the problem!

Greetings
Chris

**Comment by DimitrisGrig** [comment]
Hi @Chrizey91, thank you for your message.

I utilized the tool again with the `-D` parameter as you asked. The output can be seen in the following image:

[https://i.imgur.com/YUNa3fn.png](url)

Best,
Dimitris

**Comment by Chrizey91** [comment]
Ha, the notification-system worked this time!

And I think I found the problem. 

*Tl;Dr:*
You forgot the `-rep X` parameter. Simply

*Longer version:*
iTiSS is able to merge the read-counts of multiple BAM-files and conditions of CIT-files. However, sometimes you want to merge only a few specific conditions/files. With the `-rep` parameter, you can mark the conditions you want to merge. E.g.: if you have a CIT-file with three conditions, or provide 3 BAM files but only want to include/merge condition/file 1 and 3, the parameter would look like this: `X_X`. `X` marks the file/condition to use and the `_` the condition/file to ignore.

I will add a helper message in the next update to prevent this kind of confusion if the parameter is left out.

Greetings

**Comment by DimitrisGrig** [comment]
I still had no luck with the `-rep X` parameter. You can see the output on the image below:

[https://i.imgur.com/6qngzeN.png](https://i.imgur.com/6qngzeN.png)

Best,
Dimitris

**Comment by Chrizey91** [comment]
The only possible way for this error to appear is if you provide a negative chromosome size in your `GRCh38chrom.sizes`-file. Could you paste the input of said file here?

If that is not the case, would it be possible if you could send me your chrom.sizes file as well as the .bam file in order for me to reproduce the issue (e.g. uploading it on google drive and pasting the link here etc.)?

Greetings

**Comment by DimitrisGrig** [comment]
I uploaded the files here: [Link](https://drive.google.com/drive/folders/1t29hCj-rhZ0VJxlLN4V7Mq0GXGs4JAHt?usp=sharing)

`GRCh38chrom.sizes` is obtained from UCSC and the bam is from fantom5 repository.

If you utilize iTiSS it would help me a lot if you share the results with me!

Thank you for your time and effort,
Dimitris

**Comment by Chrizey91** [comment]
Thank you for providing the data. I was able to pinpoint the problem!

It seems that iTiSS does not like the `chr1, chr2,...` format. but rather wants to have `1, 2, ....`.
This is obviously bad, and I will fix this in the next version of iTiSS!

For now, in the `chrom.sizes`-file, you need to omit the `chr` prefix. I attached an appropriately altered version of your file for you to use. However, GitHub does not allow the file to end on `.sizes`, so I added a `.txt` to it. Simply delete the `.txt`-extension, if you use it.

Sorry for the inconvenience and thanks for helping me find this bug!

Greetings
[GRCh38chrom.sizes.txt](https://github.com/erhard-lab/iTiSS/files/6576892/GRCh38chrom.sizes.txt)

**Comment by DimitrisGrig** [comment]
Thank @Chrizey91!
I can confirm that was the issue. I kindly suggest using the prefix `-chr` as it complies with the default chromosome size files in the repositories (e.g. UCSC).

Best,
Dimitris

**Comment by Chrizey91** [comment]
I am very happy to here that I could help you!

With the next update iTiSS will simply support both ways. So it shouldn't matter anymore if you write `chr1` or just `1`. :)

Greetings
